tests/k8st/tests/test_bgp_advert.py

```diff
@@ -0,0 +1,217 @@
+        subprocess.check_call("docker exec kube-node-extra apt install -y bird", shell=True)
+        subprocess.check_call("docker exec kube-node-extra mkdir /run/bird", shell=True)
+        with open('bird.conf', 'w') as birdconfig:
+            birdconfig.write(bird_conf)
+        subprocess.check_call("docker cp bird.conf kube-node-extra:/etc/bird/bird.conf", shell=True)
+        subprocess.check_call("rm bird.conf", shell=True)
+        subprocess.check_call("docker exec kube-node-extra service bird restart", shell=True)
+
+        # # Create nginx deployment and service
+        # subprocess.check_call("kubectl create deploy nginx --image=nginx", shell=True)
+        # subprocess.check_call("kubectl create svc clusterIP nginx --tcp 80:80", shell=True)
+        cluster = self.k8s_client()
+        cluster.create_namespace(client.V1Namespace(metadata=client.V1ObjectMeta(name="bgp-test")))
+        container = client.V1Container(
+            name="nginx",
+            image="nginx:1.7.9",
+            ports=[client.V1ContainerPort(container_port=80)])
+        # Create and configure a spec section
+        template = client.V1PodTemplateSpec(
+            metadata=client.V1ObjectMeta(labels={"app": "nginx"}),
+            spec=client.V1PodSpec(containers=[container]))
+        # Create the specification of deployment
+        spec = client.ExtensionsV1beta1DeploymentSpec(
+            replicas=2,
+            template=template)
+        # Instantiate the deployment object
+        nginx_deployment = client.ExtensionsV1beta1Deployment(
+            api_version="extensions/v1beta1",
+            kind="Deployment",
+            metadata=client.V1ObjectMeta(name="nginx"),
+            spec=spec)
+        api_response = client.ExtensionsV1beta1Api().create_namespaced_deployment(
+            body=nginx_deployment,
+            namespace="bgp-test")
+        _log.debug("Deployment created. status='%s'" % str(api_response.status))
+
+        service = client.V1Service(
+            metadata=client.V1ObjectMeta(
+                name="nginx",
+                labels={"name": "nginx"},
+            ),
+            spec={
+                "ports": [{"port": 80}],
+                "selector": {"app": "nginx"},
+            }
+        )
+        api_response = cluster.create_namespaced_service(
+            body=service,
+            namespace="bgp-test",
+        )
+        _log.debug("Service created. status='%s'" % str(api_response.status))
+
+        # # Modify calico/node to run :master tag and set CALICO_STATIC_ROUTES=10.96.0.0/12
+
+        config.load_kube_config(os.environ.get('KUBECONFIG'))
+        api = client.AppsV1Api(client.ApiClient())
+        node_ds = api.read_namespaced_daemon_set("calico-node", "kube-system", exact=True, export=True)
+        for container in node_ds.spec.template.spec.containers:
+            if container.name == "calico-node":
+                container.image = "calico/node:latest-amd64"
```

Comment:
Feels like changing the calico/node image to be the one just built should be a thing that happens for ALL tests in here, not just this one.  

Consider moving it to cluster setup steps.